text_file_functions.h

```diff
@@ -9,5 +9,214 @@
 #include <QDebug>
 #include <QColor>
 #include <QSettings>
+#include <QDir>
+#include <QStandardPaths>
+#include <QRegExp>
+
+/// A singleton designed to handle the config.ini file.
+/// Call upon it with TextFileHandler::getInstance().
+class TextFileHandler
+{
+  private:
+    TextFileHandler() {}
+
+    // Isn't this ironic? To eliminate pointers to the AO app,
+    // I must still keep one.
+    AOApplication *ao_app;
+
+    QString current_theme = "default";
+
+  public:
+    static TextFileHandler& getInstance();
+    TextFileHandler(TextFileHandler const&) = delete;
+    void operator=(TextFileHandler const&) = delete;
+
+    QSettings *configini;
+
+    // Gives the pointer to the AO app to it.
+    // Should be called right as the AO app itself is created.
+    void give_ao_app_pointer(AOApplication *p_ao_app);
```

Comment:
I would have done this differently - make a `createInstance` function that creates the singleton ahead of time, and then you don't have to worry about setting individual values at start time.